Sources/NIOTransportServices/NIOTSListenerBootstrap.swift

```diff
@@ -377,4 +377,225 @@ public final class NIOTSListenerBootstrap {
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+extension NIOTSListenerBootstrap {
+    /// Bind the `NIOTSListenerChannel` to `host` and `port`.
+    ///
+    /// - Parameters:
+    ///   - host: The host to bind on.
+    ///   - port: The port to bind on.
+    ///   - serverBackpressureStrategy: The back pressure strategy used by the server socket channel.
+    ///   - channelInitializer: A closure to initialize the channel. The return value of this closure is returned from the `bind`
+    ///     method.
+    /// - Returns: The result of the channel initializer.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    public func bind<Output: Sendable>(
+        host: String,
+        port: Int,
+        serverBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
+        childChannelInitializer: @escaping @Sendable (Channel) -> EventLoopFuture<Output>
+    ) async throws -> NIOAsyncChannel<Output, Never> {
+        let validPortRange = Int(UInt16.min)...Int(UInt16.max)
+        guard validPortRange.contains(port) else {
+            throw NIOTSErrors.InvalidPort(port: port)
+        }
+
+        return try await self.bind0(
+            serverBackpressureStrategy: serverBackpressureStrategy,
+            childChannelInitializer: childChannelInitializer,
+            registration: { (serverChannel, promise) in
+                serverChannel.register().whenComplete { result in
+                    switch result {
+                    case .success:
+                        do {
+                            // NWListener does not actually resolve hostname-based NWEndpoints
+                            // for use with requiredLocalEndpoint, so we fall back to
+                            // SocketAddress for this.
+                            let address = try SocketAddress.makeAddressResolvingHost(host, port: port)
+                            serverChannel.bind(to: address, promise: promise)
+                        } catch {
+                            promise.fail(error)
+                        }
+                    case .failure(let error):
+                        promise.fail(error)
+                    }
+                }
+            }
+        ).get()
+    }
+
+    /// Bind the `NIOTSListenerChannel` to `address`.
+    ///
+    /// - Parameters:
+    ///   - address: The `SocketAddress` to bind on.
+    ///   - serverBackpressureStrategy: The back pressure strategy used by the server socket channel.
+    ///   - channelInitializer: A closure to initialize the channel. The return value of this closure is returned from the `bind`
+    ///     method.
+    /// - Returns: The result of the channel initializer.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    public func bind<Output: Sendable>(
+        to address: SocketAddress,
+        serverBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
```

Comment:
spelling: should this be `serverBackPressureStrategy`? Please audit this elsewhere too.